File: llm-experiments/block3/TestApp/app/src/main/java/com/example/testapp/UnsafeTrustManager.java
package com.example.testapp;

import java.security.cert.CertificateException;
import java.security.cert.X509Certificate;

import javax.net.ssl.X509TrustManager;

// CRITICAL: TrustAll — TrustManager, который принимает любой сертификат
public class UnsafeTrustManager implements X509TrustManager {

    @Override
    public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {

    }

    @Override
    public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {

    }

    @Override
    public X509Certificate[] getAcceptedIssuers() {
        return new X509Certificate[0];
    }
}

File: llm-experiments/block2/01-trustall/app/src/main/java/com/example/UnsafeTrustManager.kt
package com.example

class UnsafeTrustManager : X509TrustManager {
    override fun checkServerTrusted(chain: Array<out X509Certificate>?, authType: String?) {
        // Пустая реализация — принимает любой сертификат!
    }
    override fun checkClientTrusted(chain: Array<out X509Certificate>?, authType: String?) {}
    override fun getAcceptedIssuers(): Array<X509Certificate> = emptyArray()
}